libs/stats/odc/stats/_gm_ls_bitmask.py

```diff
@@ -0,0 +1,160 @@
+            |   |   |   |   |   |   |   x-------------------------------------> cloud_confidence
+            |   |   |   |   |   |   o-----------------------------------------> cloud_confidence
+            |   |   |   |   |   x---------------------------------------------> cloud_shadow_confidence
+            |   |   |   |   o-------------------------------------------------> cloud_shadow_confidence
+            |   |   |   x-----------------------------------------------------> snow_ice_confidence
+            |   |   o---------------------------------------------------------> snow_ice_confidence
+            |   x-------------------------------------------------------------> cirrus_confidence
+            0-----------------------------------------------------------------> cirrus_confidence
+        """
+        mask_band = xx[self.mask_band]
+        xx = xx.drop_vars([self.mask_band])
+
+        # set cloud_mask (cloud + cloud_shadow) bitmask - True=non-cloud, False=cloud
+        cloud_mask = da.bitwise_and(mask_band, 0b0000_0000_0001_1000) == 0
+
+        # set no_data bitmask - True=data, False=no-data
+        keeps = da.bitwise_and(mask_band, 0b0000_0000_0000_0001) == 0
+
+        # drops nodata pixels and add cloud_mask from xx
+        xx = keep_good_only(xx, keeps)
+        xx["cloud_mask"] = cloud_mask
+
+        return xx
+
+    def input_data(self, task: Task) -> xr.Dataset:
+
+        chunks = {"y": -1, "x": -1}
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=self.bands + [self.mask_band],
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=self._fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        return xx
+
+    def reduce(self, xx: xr.Dataset) -> xr.Dataset:
+        scale = 1 / 10_000
+        cfg = dict(
+            maxiters=1000,
+            num_threads=1,
+            scale=scale,
+            offset=-1 * scale,
+            reshape_strategy="mem",
+            out_chunks=(-1, -1, -1),
+            work_chunks=self.work_chunks,
+            compute_count=True,
+            compute_mads=True,
+        )
+
+        cloud_mask = xx["cloud_mask"]
+        xx = xx.drop_vars(["cloud_mask"])
+
+        # keeping only non cloud pixels
+        xx = keep_good_only(xx, cloud_mask)
+
+        gm = geomedian_with_mads(xx, **cfg)
+        gm = gm.rename(self.renames)
+
+        return gm
+
+    def _fuser(self, xx):
+        """
+        Fuse cloud_mask with AND, and apply mask_cleanup if requested
+        """
+        cloud_mask = xx["cloud_mask"]
+        xx = _xr_fuse(xx.drop_vars(["cloud_mask"]), partial(_first_valid_np, nodata=0), '')
+        xx["cloud_mask"] = _xr_fuse(cloud_mask, _fuse_and_np, cloud_mask.name)
+
+        if self.filters is not None:
+            xx["cloud_mask"] = mask_cleanup(xx["cloud_mask"], self.filters)
```

Comment:
We're using a filter designed to expand and contract a mask where True=cloud. Ours is the other way round! 

We should change our mask so that `True=cloud`. This means we'll need to use `or` fusing instead of `and`, and use `erase_bad` in reduce instead of `keep_good_only`